src/ui/egg-gacha-ui-handler.ts

```diff
@@ -387,142 +356,187 @@ export default class EggGachaUiHandler extends MessageUiHandler {
-    }
-    if (!eggs) {
-      eggs = [];
-      for (let i = 1; i <= pullCount; i++) {
-        const eggOptions: IEggOptions = {
-          pulled: true,
-          sourceType: this.gachaCursor,
-        };
-
-        // Before creating the last egg, check if the guaranteed egg tier was already generated
-        // if not, override the egg tier
-        if (i === pullCount) {
-          const guaranteedEggTier = this.getGuaranteedEggTierFromPullCount(pullCount);
-          if (!eggs.some(egg => egg.tier >= guaranteedEggTier) && guaranteedEggTier !== EggTier.COMMON) {
-            eggOptions.tier = guaranteedEggTier;
-          }
+      // Before creating the last egg, check if the guaranteed egg tier was already generated
+      // if not, override the egg tier
+      if (i === pullCount) {
+        const guaranteedEggTier = this.getGuaranteedEggTierFromPullCount(pullCount);
+        if (guaranteedEggTier !== EggTier.COMMON && !eggs.some(egg => egg.tier >= guaranteedEggTier)) {
+          eggOptions.tier = guaranteedEggTier;
         }
+      }
 
-        const egg = new Egg(eggOptions);
-        eggs.push(egg);
+      const egg = new Egg(eggOptions);
+      eggs.push(egg);
+    }
+    // Shuffle the eggs in case the guaranteed one got added as last egg
+    return randSeedShuffle(eggs);
+  }
+
+  /**
+   * Handle pulling eggs from the gacha machine; plays the animations, adds the eggs, and saves game data
+   * @param pullCount - The number of eggs to pull
+   */
+  async pull(pullCount = 0): Promise<void> {
+    if (Overrides.EGG_GACHA_PULL_COUNT_OVERRIDE) {
+      pullCount = Overrides.EGG_GACHA_PULL_COUNT_OVERRIDE;
+    }
+
+    // Set the eggs
+    const eggs = this.pullEggs(pullCount);
+
+    this.eggGachaOptionsContainer.setVisible(false);
+    this.setTransitioning(true);
+
+    const saveSuccess = await (globalScene.currentBattle
+      ? globalScene.gameData.saveAll(true, true, true)
+      : globalScene.gameData.saveSystem()
+    ).then(success => {
+      if (!success) {
+        globalScene.reset(true);
+        return false;
       }
-      // Shuffle the eggs in case the guaranteed one got added as last egg
-      eggs = randSeedShuffle<Egg>(eggs);
-
-      (globalScene.currentBattle
-        ? globalScene.gameData.saveAll(true, true, true)
-        : globalScene.gameData.saveSystem()
-      ).then(success => {
-        if (!success) {
-          return globalScene.reset(true);
-        }
-        doPull();
-      });
+      return true;
+    });
+
+    if (!saveSuccess) {
       return;
     }
 
-    doPull();
+    const gachaContainer = this.gachaContainers[this.gachaCursor];
+    for (let i = 0; i < pullCount; ++i) {
+      if (this.transitionCancelled) {
+        break;
+      }
+      const eggSprite = globalScene.add.sprite(127, 75, "egg", `egg_${eggs[i].getKey()}`).setScale(0.5);
+      gachaContainer.addAt(eggSprite, 2);
+      await this.doPullAnim(eggSprite, i).then(() => gachaContainer.remove(eggSprite, true));
```

Comment:
Since you're using `await` doesn't that mean you can do this?
```suggestion
      await this.doPullAnim(eggSprite, i);
      gachaContainer.remove(eggSprite, true);
```
I thought `await` was an alternative to `x.then()`.

Reply:
It is, sort of. I actually meant for this to be a `.finally()` not a `.then()` just so that the egg sprites are _always removed_ regardless of an error that may or may not have been raised during the method.
This is because I want to aggressively remove the egg sprite that was added.